agent_app_sample_code/00_shared_config.py

```diff
@@ -0,0 +1,95 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC ## Shared configuration
+# MAGIC
+# MAGIC This notebook initializes a `AgentCookbookConfig` Pydantic class to define parameters that are shared between the cookbook notebooks:
+# MAGIC - Unity Catalog schema 
+# MAGIC - MLflow Experiment to track versions of the Agent and their associated quality/cost/latency evaluation results
+# MAGIC - Unity Catalog model that stores versions of the Agent's code/config that are deployed
+# MAGIC - Evaluation Set Delta Table
+# MAGIC
+# MAGIC This notebook does the following:
+# MAGIC 1. Creates the UC catalog/schema if they don't exist
+# MAGIC 2. Serializes the configuration to `config/cookbook_config.yaml` so other notebooks can load it
+# MAGIC
+# MAGIC **Important: We suggest starting from a fresh clone of the cookbook if you need to change this configuration.**
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC **Important note:** Throughout this notebook, we indicate which cells you:
+# MAGIC - ✅✏️ *should* customize - these cells contain config settings to change
+# MAGIC - 🚫✏️ *typically will not* customize - these cells contain boilerplate code required to validate / save the configuration
```

Comment:
Why not put the things that shouldn't be customized in python files?

Reply:
We can move more into the python files, but i tried to strike a balance of not hiding code that the user should be aware of.  for example, i made a cell with mlflow logging boilerplate that shouldn't be modified but i also didnt want to hide it from the user since we got feedback before that hiding the details of how model logging works was confusing

Reply:
Ack - probably out of scope for this PR but I'd prefer some better logging, output rendering and maybe the ability to change the config inline using an input Ipywidget or similar. Fewer notebook cells, less to think about

Reply:
I kinda agree with @FMurray here, as a reader these notes are quite helpful but they seem hard for us to maintain. I wonder if we could recommend folks to use IDEs to view the cookbook (that's what I'm doing right now as I review & try out changes to this PR), which has better jump-to-definition support for things like easily reading the code in Python modules, if readers are interested - that way we can try to keep maintainability but avoid the "hidden magic code" issue

Reply:
TBH though, there are other things that I think affect maintainability more, so fine to try this to start, it is quite helpful